compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -486,26 +486,26 @@ impl Printer<'tcx> for SymbolMangler<'tcx> {
         predicates: &'tcx ty::List<ty::Binder<'tcx, ty::ExistentialPredicate<'tcx>>>,
     ) -> Result<Self::DynExistential, Self::Error> {
         for predicate in predicates {
-            self = self.in_binder(&predicate, |mut cx, predicate| {
-                match predicate {
-                    ty::ExistentialPredicate::Trait(trait_ref) => {
+            match predicate.as_ref().skip_binder() {
```

Comment:
Can you add a comment about why we jump though these hoops?

Reply:
How's this? da404b52d91b5644f2a6490ee0f22021e985bb22

Reply:
I think I'd prefer a little bit better explanation. Something along the lines of "Symbol mangling does not output binder information for projection predicates. Instead, it assumes that the binders are the same as on the trait ref. See issue #83611."

Reply:
Sure, that seems good. My comment doesn't have much information because I'm new to trait solving/binders/etc. :)

Reply:
Still waiting on this comment :)

Reply:
Added :)